scalajslib/src/mill/scalajslib/ScalaJSModule.scala

```diff
@@ -73,22 +75,31 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
   def scalaJSToolsClasspath = T { scalaJSWorkerClasspath() ++ scalaJSLinkerClasspath() }
 
-  def fastOpt = T {
-    linkTask(FastOpt)()
+  def fastLinkJS: Target[Report] = T {
+    linkTask(optimize = false, legacy = false)()
   }
 
-  def fullOpt = T {
-    linkTask(FullOpt)()
+  def fullLinkJS: Target[Report] = T {
+    linkTask(optimize = true, legacy = false)()
   }
 
-  private def linkTask(mode: OptimizeMode): Task[PathRef] = T.task {
-    link(
-      worker = ScalaJSWorkerApi.scalaJSWorker(),
+  def fastOpt: Target[PathRef] = T {
+    linkTask(optimize = false, legacy = true)().publicModules.head.jsFile
+  }
+
+  def fullOpt: Target[PathRef] = T {
+    linkTask(optimize = true, legacy = true)().publicModules.head.jsFile
+  }
+
+  private def linkTask(optimize: Boolean, legacy: Boolean): Task[Report] = T.task {
```

Comment:
Can we have a more descriptive name than `legacy`? E.g. it this is indicating pre-1.2 versions, we might just name the parameter that way.

Reply:
Actually legacy means "Use the `PathOutputFile` linking mode instead of `PathOutputDirectory`" which is the one used by the `fastOptJS` (which is the command that Scala.js 1.3 deprecated).

So in Scala.js 1.2- it always links using `PathOutputFile` while in Scala.js 1.3+ the `legacy` flag decides if you want to use the output file or the directory.

Reply:
What about `returnPathIsFile`?

Reply:
Since we polyfill the new API we return a `Report` even in legacy mode with the same structure.
I think `legacy` is a good enough name. Also it is used only in private apis so users don't need to know about it.

Reply:
To me the name `legacy` is as good as `foo`. The switch clearly does something, otherwise we could just leave it out. Even when this is our internal API, we need to maintain it. If I have issues now, how much fun will have my successor later?

What about `forceOutJs`? We automatically use `out.js` for older Scala.js versions, but what we set this to `true`, we explicitly want to refer to a `out.js`.